tests/integration/_experimental/object_detection/test_evaluator_single_class_fixed.py

```diff
@@ -0,0 +1,506 @@
+# Copyright 2021-2023 Kolena Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from typing import Tuple
+
+import pytest
+
+from kolena.workflow.annotation import LabeledBoundingBox
+from kolena.workflow.annotation import ScoredLabeledBoundingBox
+from kolena.workflow.plot import Curve
+from kolena.workflow.plot import CurvePlot
+from kolena.workflow.plot import Plot
+from tests.integration.helper import fake_locator
+from tests.integration.helper import with_test_prefix
+
+object_detection = pytest.importorskip("kolena._experimental.object_detection")
+GroundTruth = object_detection.GroundTruth
+Inference = object_detection.Inference
+TestCase = object_detection.TestCase
+TestSample = object_detection.TestSample
+TestSuite = object_detection.TestSuite
+ThresholdConfiguration = object_detection.ThresholdConfiguration
+ThresholdStrategy = object_detection.ThresholdStrategy
+TestCaseMetricsSingleClass = object_detection.TestCaseMetricsSingleClass
+TestSampleMetricsSingleClass = object_detection.TestSampleMetricsSingleClass
+TestSuiteMetrics = object_detection.TestSuiteMetrics
+
+
+TEST_CASE_NAME = "single class OD test"
+TEST_CASE = TestCase(with_test_prefix(TEST_CASE_NAME + " case"))
+TEST_SUITE = TestSuite(with_test_prefix(TEST_CASE_NAME + " suite"))
+
+
+TEST_DATA: List[Tuple[TestSample, GroundTruth, Inference]] = [
+    (
+        TestSample(locator=fake_locator(112, "OD")),
+        GroundTruth(
+            bboxes=[
+                LabeledBoundingBox((1, 1), (2, 2), "a"),
+                LabeledBoundingBox((3, 3), (4, 4), "a"),
+                LabeledBoundingBox((5, 5), (6, 6), "a"),
+                LabeledBoundingBox((7, 7), (8, 8), "d"),
```

Comment:
should this be class "a"?

Reply:
This test checks that more than one label is allowed. In the case where a user puts multiple labels into the single class workflow, it will not differentiate labels for matching.

Reply:
Gotcha. Since the rest of the test cases had the same label, it wasn't too clear. Can we leave a comment mentioning the intention behind using different labels here?

Reply:
Yeah, that would definitely be good if we come back to these tests in the future!